src/utils/index.ts

```diff
@@ -1363,3 +1364,41 @@ export function middlewareProposalToTxTag(bytes: string, context: Context): TxTa
     callId: methodName as CallIdEnum,
   });
 }
+
+/**
+ * @hidden
+ */
+export function middlewareProposalToProposalDetails(
+  proposal: Proposal,
+  context: Context
+): ProposalDetails {
+  const {
+    proposer: proposerAddress,
+    createdAt,
+    url: discussionUrl,
+    description,
+    coolOffEndBlock,
+    endBlock,
+    proposal: rawProposal,
+    lastState,
+    lastStateUpdatedAt,
+    totalVotes,
+    totalAyesWeight,
+    totalNaysWeight,
+  } = proposal;
+
+  return {
+    proposerAddress,
+    createdAt: new BigNumber(createdAt),
+    discussionUrl,
+    description,
+    coolOffEndBlock: new BigNumber(coolOffEndBlock),
+    endBlock: new BigNumber(endBlock),
+    transaction: rawProposal ? middlewareProposalToTxTag(rawProposal, context) : null,
```

Comment:
`middlewareProposalToTxTag` should be renamed to `transactionHexToTxTag` to avoid confusion